handlers.go

```diff
@@ -281,55 +567,262 @@ func deleteProfileHandler(w *Web) {
-rm clients/{{$.Profile.ID}}.conf
-`
-	output, err := bash(script, struct {
-		Profile Profile
-	}{
-		profile,
-	})
-	if err != nil {
-		logger.Warnf("delete profile failed %s %s", err, output)
-		w.Redirect("/profiles/delete?error=removeprofile")
+	if err := deleteProfile(profile); err != nil {
+		logger.Errorf("delete profile failed: %s", err)
+		w.Redirect("/profile/delete?error=deleteprofile")
 		return
 	}
-
-	if err := config.DeleteProfile(profile.ID); err != nil {
-		panic(err)
+	if profile.UserID != "" {
+		w.Redirect("/user/edit/%s?success=deleteprofile", profile.UserID)
+		return
 	}
-	w.Redirect("/?success=removeprofile")
+	w.Redirect("/?success=deleteprofile")
 }
 
 func indexHandler(w *Web) {
-	profiles := config.ListProfiles()
+	if w.User.ID != "" {
+		w.TargetProfiles = config.ListProfilesByUser(w.User.ID)
+	}
+	if w.Admin {
+		w.Profiles = config.ListProfilesByUser("")
+		w.Users = config.ListUsers()
+	} else {
+		w.Profiles = config.ListProfilesByUser(w.User.ID)
+	}
+	w.HTML()
+}
+
+func RoundUp(input float64, places int) (newVal float64) {
+	var round float64
+	pow := math.Pow(10, float64(places))
+	digit := pow * input
+	round = math.Ceil(digit)
+	newVal = round / pow
+	return
+}
+
+func ByteFormat(inputNum float64, precision int) string {
+
+	if precision <= 0 {
+		precision = 1
+	}
+
+	var unit string
+	var returnVal float64
+
+	if inputNum >= 1000000000000 {
+		returnVal = RoundUp((inputNum / 1099511627776), precision)
+		unit = " TiB" // terrabyte
+	} else if inputNum >= 1000000000 {
+		returnVal = RoundUp((inputNum / 1073741824), precision)
+		unit = " GiB" // gigabyte
+	} else if inputNum >= 1000000 {
+		returnVal = RoundUp((inputNum / 1048576), precision)
+		unit = " MiB" // megabyte
+	} else if inputNum >= 1000 {
+		returnVal = RoundUp((inputNum / 1024), precision)
+		unit = " KiB" // kilobyte
+	} else {
+		returnVal = inputNum
+		unit = " B" // byte
+	}
+
+	return strconv.FormatFloat(returnVal, 'f', precision, 64) + unit
+}
+
+func TimeDiff(a, b time.Time) (year, month, day, hour, min, sec int) {
+	if a.Location() != b.Location() {
+		b = b.In(a.Location())
+	}
+	if a.After(b) {
+		a, b = b, a
+	}
+	y1, M1, d1 := a.Date()
+	y2, M2, d2 := b.Date()
+
+	h1, m1, s1 := a.Clock()
+	h2, m2, s2 := b.Clock()
+
+	year = int(y2 - y1)
+	month = int(M2 - M1)
+	day = int(d2 - d1)
+	hour = int(h2 - h1)
+	min = int(m2 - m1)
+	sec = int(s2 - s1)
+
+	// Normalize negative values
+	if sec < 0 {
+		sec += 60
+		min--
+	}
+	if min < 0 {
+		min += 60
+		hour--
+	}
+	if hour < 0 {
+		hour += 24
+		day--
+	}
+	if day < 0 {
+		// days in month:
+		t := time.Date(y1, M1, 32, 0, 0, 0, 0, time.UTC)
+		day += 32 - t.Day()
+		month--
+	}
+	if month < 0 {
+		month += 12
+		year--
+	}
+
+	return
+}
+
+func Short(s string, i int) string {
+	runes := []rune(s)
+	if len(runes) > i {
+		return string(runes[:i])
+	}
+	return s
+}
+
+func statusHandler(w *Web) {
+	wg_dump, err := exec.Command("wg", "show", "all", "dump").Output()
```

Comment:
All other handlers use wg0, displayed clients should be only from wg0 to.
```go
	wg_dump, err := exec.Command("wg", "show", "wg0", "dump").Output()
```

Reply:
I did not pay attention to the rest of the code and missed that you are using the interface name later. But is it really useful when we can't specify an interface name when creating clients? Maybe it would be better to allow global per subspace configuration of interface name to enable running multiple containers at the same time.